packages/sitecore-jss/src/i18n/graphql-dictionary-service.ts

```diff
@@ -158,53 +108,42 @@ export class GraphQLDictionaryService extends DictionaryServiceBase {
       return cachedValue;
     }
 
-    const client = new GraphQLRequestClient(this.options.endpoint, {
-      apiKey: this.options.apiKey,
-      debugger: debug.dictionary,
-    });
-    if (!this.options.rootItemId) {
-      debug.dictionary('fetching site root for %s %s', language, this.options.siteName);
-      this.options.rootItemId = await getAppRootId(client, this.options.siteName, language);
+    // If the caller does not specify a root item ID, then we try to figure it out
+    const rootItemId =
+      this.options.rootItemId ||
+      (await getAppRootId(this.graphQLClient, this.options.siteName, language));
+
+    if (!rootItemId) {
+      throw new Error(queryError);
     }
 
     debug.dictionary('fetching dictionary data for %s %s', language, this.options.siteName);
-    const results = await this.getDictionaryPhrases(client, language);
-    this.setCacheValue(cacheKey, results);
-    return results;
-  }
-
-  /**
-   * Gets dictionary phrases
-   * @param {GraphQLRequestClient} client that fetches data from a GraphQL endpoint.
-   * @param {string} language
-   * @returns dictionary phrases
-   */
-  async getDictionaryPhrases(
-    client: GraphQLRequestClient,
-    language: string
-  ): Promise<DictionaryPhrases> {
-    const results: DictionaryPhrases = {};
-    let hasNext = true;
-    let after = '';
-
-    while (hasNext) {
-      const fetchResponse = await client.request<DictionaryQueryResult>(query, {
-        // `search` query only works with lowercase GUIDs
-        rootItemId: this.options.rootItemId?.toLowerCase(),
+    const phrases: DictionaryPhrases = {};
+    await this.searchService
+      .fetch(query, {
+        rootItemId,
         language,
-        dictionaryEntryTemplateId: SitecoreTemplateId.DictionaryEntry,
+        templates: this.options.dictionaryEntryTemplateId || SitecoreTemplateId.DictionaryEntry,
         pageSize: this.options.pageSize,
-        after,
-      });
-
-      fetchResponse.search.dictionaryPhrases.forEach((dictionaryPhrase) => {
-        results[dictionaryPhrase.key.value] = dictionaryPhrase.phrase.value;
+      })
+      .then((results) => {
+        results.forEach((item) => (phrases[item.key.value] = item.phrase.value));
       });
 
-      hasNext = fetchResponse.search.pageInfo.hasNext;
-      after = fetchResponse.search.pageInfo.endCursor;
-    }
+    this.setCacheValue(cacheKey, phrases);
+    return phrases;
+  }
 
-    return results;
+  /**
+   * Gets a GraphQL client that can make requests to the API. Uses graphql-request as the default
+   * library for fetching graphql data (@see GraphQLRequestClient). Override this method if you
+   * want to use something else.
+   * @returns {GraphQLClient} implementation
+   */
+  protected getGraphQLClient(): GraphQLClient {
```

Comment:
`GraphQLLayoutService` needs the new signature / implementation for `getGraphQLClient()` applied:
https://github.com/Sitecore/jss/blob/feature/467450-2/packages/sitecore-jss/src/layout/graphql-layout-service.ts#L72 

It's still private and returning `GraphQLRequestClient` instead of `GraphQLClient`

Reply:
Done. Good catch.